src/OSPSuite.Core/Services/CurveChartExportTask.cs

```diff
@@ -46,14 +53,17 @@ public void ExportToExcel(CurveChart chart)
          visibleCurves.Each(curve => dataColumnCache[curve.yData] = curve);
 
          //Base grid are added by default to the export unless the data represents an amount vs obs data. In that case, the base grid might be another column
-         var otherColumnsToExport = visibleCurves.Select(x => x.xData).Where(x => !x.IsBaseGrid());
+         var otherColumnsToExport = visibleCurves.Select(x => x.xData).Where(x => !x.IsBaseGrid()).ToList();
 
          var exportOptions = new DataColumnExportOptions
          {
             ColumnNameRetriever = col => dataColumnCache[col]?.Name ?? col.Name,
             DimensionRetriever = _dimensionFactory.MergedDimensionFor
          };
-         _dataRepositoryExportTask.ExportToExcel(dataColumnCache.Keys.Union(otherColumnsToExport), fileName, exportOptions: exportOptions);
+
+         var allColumns = dataColumnCache.Keys.Union(otherColumnsToExport);
+         var allColumnsToExport = preExportHook?.Invoke(allColumns) ?? allColumns;
+         _dataRepositoryExportTask.ExportToExcel(allColumnsToExport, fileName, exportOptions: exportOptions);
```

Comment:
there is no way that we can modify the order of the columns here as we do not know what is what. It needs to be done by the caller. I am adding this hook to enable this